### PR TITLE
Refactor k8senv adapter to generate workload info.

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR is part of the general refactoring of the `kubernetesenv` Mixer adapter (an APA) to match the expectations of the updated attributes model, in particular Istio workload metadata derivation.

In particular, this PR:

- adds k8s api watches for all versions of `ReplicaSets`, to enable derivation of root controller for pods
- adds workload and owner fields to the output template for the adapter
- removes all notions around deriving service(s) for source and destination
- removes ingress-related filtering for derivations (this should be managed via config moving forward)
- removes vestiges of `origin` related metadata

Not included in this PR:

- `metadata`-related derivation and template fields
- derivation of `destination.service.*` data (this should be provided by the proxy via Pilot)
- handling for cronjob -> job relationship in root controller derivation
- full configuration update to translate the output template into attributes for downstream usage (this likely should be a distinct PR).

Reviewers, please:
- ignore the large amount of generated file changes
- focus on the changes to `cache.go` and to the test inputs/outputs in `kubernetesenv_test.go`

I would like feedback on the data in the output struct (to ensure we agree on data formats, etc.) and on the overall approach to deriving workload information.

### CODE REVIEW

File: mixer/adapter/kubernetesenv/cache.go [hunk 1]
```diff
@@ -70,17 +84,73 @@ func newCacheController(clientset kubernetes.Interface, refreshDuration time.Dur
...
+				},
+			},
+			&appsv1.ReplicaSet{},
+			refreshDuration,
+			cache.Indexers{},
+		),
+		appsv1beta2RS: cache.NewSharedIndexInformer(
+			&cache.ListWatch{
+				ListFunc: func(opts metav1.ListOptions) (runtime.Object, error) {
+					return clientset.AppsV1beta2().ReplicaSets(namespace).List(opts)
```
Comment (kyessenov) [inline mixer/adapter/kubernetesenv/cache.go]: Shouldn't k8s auto convert v1beta2 to v1 for the same kind?

Comment (douglas-reid) [reply mixer/adapter/kubernetesenv/cache.go]: does it?  If so, this indexer can be removed. I wasn't sure.

**Comment by douglas-reid** [comment]
Reviewers, PTAL. I've added the required ClusterRoleBinding bits and fleshed-out the YAML config for the `attributes.kubernetes` instances to generate the full attribute set.

Running in a test cluster, I observed the following attribute output:
```
2018-06-07T17:26:13.731294Z	debug	api	Attribute Bag: 
check.cache_hit               : true
connection.mtls               : false
context.protocol              : http
context.reporter.local        : true
context.reporter.uid          : kubernetes://reviews-v2-55694f4445-6fm7c.default
destination.ip                : [0 0 0 0 0 0 0 0 0 0 255 255 10 0 0 40]
destination.labels            : {destination.labels map[pod-template-hash:1125090001 version:v2 app:reviews] 0xc4217c5800}
destination.port              : 9080
destination.service           : reviews.default.svc.cluster.local
destination.service.host      : reviews.default.svc.cluster.local
destination.service.name      : reviews
destination.service.namespace : default
destination.service.uid       : istio://default/services/reviews
destination.uid               : kubernetes://reviews-v2-55694f4445-6fm7c.default
quota.cache_hit               : true
request.headers               : {request.headers map[x-b3-sampled:1 x-forwarded-proto:http accept-encoding:gzip, deflate accept:*/* x-b3-traceid:2f8a5baa50d1ec6d x-b3-spanid:99c585572a6fe2a3 :authority:reviews:9080 :path:/reviews/0 x-request-id:90f18b21-7bc3-971d-bb01-ddcac12263f5 user-agent:python-requests/2.18.4 content-length:0 x-b3-parentspanid:2f8a5baa50d1ec6d :method:GET] 0xc4217c5800}
request.host                  : reviews:9080
request.method                : GET
request.path                  : /reviews/0
request.scheme                : http
request.size                  : 0
request.time                  : 2018-06-07 17:26:12.714519781 +0000 UTC
request.total_size            : 575
request.useragent             : python-requests/2.18.4
response.code                 : 200
response.duration             : 15.174084ms
response.headers              : {response.headers map[content-length:379 content-type:application/json date:Thu, 07 Jun 2018 17:26:12 GMT x-powered-by:Servlet/3.1 server:envoy :status:200 x-envoy-upstream-service-time:14 content-language:en-US] 0xc4217c5800}
response.size                 : 379
response.time                 : 2018-06-07 17:26:12.729605595 +0000 UTC
response.total_size           : 553
source.ip                     : [0 0 0 0 0 0 0 0 0 0 255 255 10 0 0 41]
source.labels                 : {source.labels map[app:productpage pod-template-hash:389943713 version:v1] 0xc4217c5800}
source.uid                    : kubernetes://productpage-v1-7dff87c57-bwtkf.default
---
destination.name              : reviews-v2-55694f4445-6fm7c
destination.namespace         : default
destination.owner             : kubernetes://apis/extensions/v1beta1/namespaces/default/deployments/reviews-v2
destination.serviceAccount    : default
destination.workload.name     : reviews-v2
destination.workload.namespace: default
destination.workload.uid      : istio://default/workloads/reviews-v2
source.name                   : productpage-v1-7dff87c57-bwtkf
source.namespace              : default
source.owner                  : kubernetes://apis/extensions/v1beta1/namespaces/default/deployments/productpage-v1
source.serviceAccount         : default
source.workload.name          : productpage-v1
source.workload.namespace     : default
source.workload.uid           : istio://default/workloads/productpage-v1
```

**Comment by douglas-reid** [comment]
Amazingly, the linter complains about an import. Running `bin/fmt.sh` puts set the import order. Running `bin/fmt.sh -c` then complains about the order that the same command set.

**Comment by douglas-reid** [comment]
I have no idea why our linters think my Gopkg.lock file is different, as I haven't touched that file. But I'm hopeful a rebase + force push fixes things.

**Comment by douglas-reid** [comment]
nope. i'm guessing this is impossible ?

```
git diff HEAD --exit-code -- Gopkg.lock vendor > /go/out/dep.diff
Makefile:245: recipe for target 'depend.diff' failed
```

**Comment by kyessenov** [comment]
Lesson #1: never trust our makefile :D

**Comment by douglas-reid** [comment]
whoa! linters passed 🎉 !

**Comment by douglas-reid** [comment]
/test istio-presubmit

one of these times, the docker push will work, i swear.

**Comment by douglas-reid** [comment]
All required tests pass. May I have a lgtm ?

File: mixer/adapter/kubernetesenv/cache.go (hunk 1 above)
Comment (kyessenov) [reply mixer/adapter/kubernetesenv/cache.go]: I don't know the answer myself. @ayj might know. Can be fixed in a follow-up.

**Comment by kyessenov** [comment]
Seeing more red than green makes me happy. Thanks.

**Pull request merged**